BASE/src/ServerTelemetryChannel/Implementation/TelemetrySerializer.cs

```diff
@@ -48,9 +50,35 @@ public virtual void Serialize(ICollection<ITelemetry> items)
                 throw new Exception("TelemetrySerializer.EndpointAddress was not set.");
             }
 
-            var transmission = new Transmission(this.EndpointAddress, items) 
-                                        { TransmissionStatusEvent = this.TransmissionStatusEvent };
+            var transmission = new Transmission(this.EndpointAddress, items)
+                                    { TransmissionStatusEvent = this.TransmissionStatusEvent };
             this.Transmitter.Enqueue(transmission);
         }
+
+        public virtual Task<bool> SerializeAsync(ICollection<ITelemetry> items, CancellationToken cancellationToken)
+        {
+            if (this.EndpointAddress == null)
+            {
+                throw new Exception("TelemetrySerializer.EndpointAddress was not set.");
```

Comment:
would this get propagated to customer FlushAsync or we handle it somewhere before throwing up to user?

Reply:
This gets propagated to customers FlushAsync. I had this test case to cover it.
```
            [TestMethod]
            public void ThrowsExceptionWhenEndpointAddressIsNull()
            {
                var serializer = new TelemetrySerializer(new StubTransmitter());
                AssertEx.Throws<Exception>(() => serializer.SerializeAsync(null, default));
            }
```
Now adding one more in TelemetryBufferTests to avoid confusion.